server/src/test/java/org/cloudfoundry/identity/uaa/oauth/UaaTokenStoreTests.java

```diff
@@ -453,5 +456,12 @@ public Object invoke(Object proxy, Method method, Object[] args) throws Throwabl
         }
     }
 
+    private static Duration givenMockedExpiration() {
+        Duration durationMock = mock(Duration.class);
+        doReturn(Instant.EPOCH).when(durationMock).subtractFrom(any(Temporal.class));
+        doReturn(Instant.now().plus(Duration.ofMinutes(5))).when(durationMock).addTo(any(Temporal.class));
+        return durationMock;
+    }
+
```

Comment:
Don't you find this complicated and convoluted compared to the simplicity and elegance of injecting the Clock, and setting the time in the test?

Look at [this change](https://github.com/cloudfoundry/uaa/pull/2291/commits/cffb1d6774ba2e673426770a2f5746c1043f1fdb) for example. We were able to get rid of the SQL statements just by pushing the clock forward 4 days.

Reply:
This mock - to me - is excatly what we need , often mock a bean but here we can use mockito also for our own data
https://www.baeldung.com/mockito-annotations, 
Changed to only 3 lines